tests/suites/test_suite_md.data

```diff
@@ -3,6 +3,7 @@ MD list
 mbedtls_md_list:
 
 MD NULL/uninitialised arguments
+depends_on:MBEDTLS_MD_C
```

Comment:
This seems a bit heavy-handed (and also slightly wrong in principle). I think instead of disabling the whole test we should just guard the parts that depend on `MD_C` inside the function definition - which is already done, quite heavily actually, so you'd probably only have to the `#endif` that's after `setup()` to after `finish()` (but leave `mbedtls_md()` unguarded).

(Also, if we wanted to guard the entire test with `MD_C`, as a matter of principle the right place to do so would be the `.function` file. Dependencies belongs in the `.data` file only when it's related to the specific arguments passed to the function. When the function depends on X no matter what arguments are passed to it, then the dependency on X belongs in the `.function` file. Doesn't actually matter in this case, but it makes a difference for functions that take arguments.)

Reply:
Ok cool, I just thought that a test involving NULL arguments didn't make sense here but I understand what you're saying.

Also thanks, I didn't know that second part. I will amend.